### ISSUE TITLE
[Paper Note] Layer Normalization, Ba+, arXiv'16

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#261]
## URL
https://arxiv.org/abs/1607.06450

## Authors
Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton

## Abstract
- Training state-of-the-art, deep neural networks is computationally expensive. One way to reduce the training time is to normalize the activities of the neurons. A recently introduced technique called batch normalization uses the distribution of the summed input to a neuron over a mini-batch of training cases to compute a mean and variance which are then used to normalize the summed input to that neuron on each training case. This significantly reduces the training time in feed-forward neural networks. However, the effect of batch normalization is dependent on the mini-batch size and it is not obvious how to apply it to recurrent neural networks. In this paper, we transpose batch normalization into layer normalization by computing the mean and variance used for normalization from all of the summed inputs to the neurons in a layer on a single training case. Like batch normalization, we also give each neuron its own adaptive bias and gain which are applied after the normalization but before the non-linearity. Unlike batch normalization, layer normalization performs exactly the same computation at training and test times. It is also straightforward to apply to recurrent neural networks by computing the normalization statistics separately at each time step. Layer normalization is very effective at stabilizing the hidden state dynamics in recurrent networks. Empirically, we show that layer normalization can substantially reduce the training time compared with previously published techniques.
# Translation (by gpt-4o-mini)
- 最先端の深層ニューラルネットワークのトレーニングは計算コストが高い。トレーニング時間を短縮する一つの方法は、ニューロンの活動を正規化することである。最近導入された手法であるバッチ正規化は、ミニバッチのトレーニングケースにおけるニューロンへの合計入力の分布を使用して平均と分散を計算し、それを用いて各トレーニングケースにおけるそのニューロンへの合計入力を正規化する。この手法はフィードフォワードニューラルネットワークにおいてトレーニング時間を大幅に短縮する。しかし、バッチ正規化の効果はミニバッチサイズに依存し、リカレントニューラルネットワークにどのように適用するかは明らかではない。本論文では、バッチ正規化をレイヤー正規化に転置し、単一のトレーニングケースにおけるレイヤー内のニューロンへの合計入力から正規化に使用する平均と分散を計算する。バッチ正規化と同様に、各ニューロンには正規化後、非線形性の前に適用される独自の適応バイアスとゲインを与える。バッチ正規化とは異なり、レイヤー正規化はトレーニング時とテスト時で全く同じ計算を行う。また、各タイムステップで正規化統計を別々に計算することでリカレントニューラルネットワークに簡単に適用できる。レイヤー正規化はリカレントネットワークにおける隠れ状態のダイナミクスを安定させるのに非常に効果的である。実証的に、レイヤー正規化は以前に発表された技術と比較してトレーニング時間を大幅に短縮できることを示す。
# Summary (by gpt-4o-mini)
- バッチ正規化の代わりにレイヤー正規化を用いることで、リカレントニューラルネットワークのトレーニング時間を短縮。レイヤー内のニューロンの合計入力を正規化し、各ニューロンに独自の適応バイアスとゲインを適用。トレーニング時とテスト時で同じ計算を行い、隠れ状態のダイナミクスを安定させる。実証的に、トレーニング時間の大幅な短縮を確認。

**Comment by AkihikoWatanabe** [comment]
解説スライド：
https://www.slideshare.net/KeigoNishida/layer-normalizationnips

**Comment by AkihikoWatanabe** [comment]
__translate: Training state-of-the-art, deep neural networks is computationally expensive. One way to reduce the training time is to normalize the activities of the neurons. A recently introduced technique called batch normalization uses the distribution of the summed input to a neuron over a mini-batch of training cases to compute a mean and variance which are then used to normalize the summed input to that neuron on each training case. This significantly reduces the training time in feed-forward neural networks. However, the effect of batch normalization is dependent on the mini-batch size and it is not obvious how to apply it to recurrent neural networks. In this paper, we transpose batch normalization into layer normalization by computing the mean and variance used for normalization from all of the summed inputs to the neurons in a layer on a single training case. Like batch normalization, we also give each neuron its own adaptive bias and gain which are applied after the normalization but before the non-linearity. Unlike batch normalization, layer normalization performs exactly the same computation at training and test times. It is also straightforward to apply to recurrent neural networks by computing the normalization statistics separately at each time step. Layer normalization is very effective at stabilizing the hidden state dynamics in recurrent networks. Empirically, we show that layer normalization can substantially reduce the training time compared with previously published techniques.
